src/core/parsers/BytecodeInstructions.js

```diff
@@ -0,0 +1,306 @@
+  },
+  { // LDC_W - LDC2_W
+    low_opcode_index: 0x13,
+    high_opcode_index: 0x14,
+    type: ImmediateShortInstruction,
+    wrap: ConstantInstruction,
+    subject_to_wide: false
+  },
+  { // GETSTATIC - PUTFIELD
+    low_opcode_index: 0xb2,
+    high_opcode_index: 0xb5,
+    type: FieldInstruction,
+    subject_to_wide: false
+  },
+  { // LDC (subject to 'wide')
+    low_opcode_index: 0x12,
+    high_opcode_index: 0x12,
+    type: ImmediateByteInstruction,
+    wrap: ConstantInstruction,
+    subject_to_wide: true
+  },
+  { // BIPUSH (subject to 'wide')
+    low_opcode_index: 0x10,
+    high_opcode_index: 0x10,
+    type: ImmediateByteInstruction,
+    wrap: PushInstruction,
+    subject_to_wide: true
+  },
+  { // SIPUSH (subject to 'wide')
+    low_opcode_index: 0x11,
+    high_opcode_index: 0x11,
+    type: ImmediateShortInstruction,
+    wrap: PushInstruction,
+    subject_to_wide: true
+  },
+  { // NEW
+    low_opcode_index: 0xbb,
+    high_opcode_index: 0xbb,
+    type: TypeInstruction,
+    subject_to_wide: false
+  },
+  { // ANEWARRAY
+    low_opcode_index: 0xbd,
+    high_opcode_index: 0xbd,
+    type: TypeInstruction,
+    subject_to_wide: false
+  },
+  { // CHECKCAST - INSTANCEOF
+    low_opcode_index: 0xc0,
+    high_opcode_index: 0xc1,
+    type: TypeInstruction,
+    subject_to_wide: false
+  },
+  { // IFEQ - JSR
+    low_opcode_index: 0x99,
+    high_opcode_index: 0xa8,
+    type: BranchInstruction,
+    subject_to_wide: false
+  },
+  { // IFNULL - IFNONNULL
+    low_opcode_index: 0xc6,
+    high_opcode_index: 0xc7,
+    type: BranchInstruction,
+    subject_to_wide: false
+  },
+  { // GOTO_W - JSR_W
+    low_opcode_index: 0xc8,
+    high_opcode_index: 0xc9,
+    type: WideBranchInstruction,
+    subject_to_wide: false
+  },
+  { // TABLESWITCH
+    low_opcode_index: 0xaa,
+    high_opcode_index: 0xaa,
+    type: TableSwitchInstruction,
+    subject_to_wide: false
+  },
+  { // LOOKUPSWITCH
+    low_opcode_index: 0xab,
+    high_opcode_index: 0xab,
+    type: LookupSwitchInstruction,
+    subject_to_wide: false
+  },
+  { // INVOKEINTERFACE
+    low_opcode_index: 0xb9,
+    high_opcode_index: 0xb9,
+    type: InvokeInterfaceInstruction,
+    subject_to_wide: false
+  },
+  { // INVOKEDYNAMIC
+    low_opcode_index: 0xba,
+    high_opcode_index: 0xba,
+    type: InvokeDynamicInstruction,
+    subject_to_wide: false
+  },
+  { // MULTIANEWARRAY
+    low_opcode_index: 0xc5,
+    high_opcode_index: 0xc5,
+    type: MultianewarrayInstruction,
+    subject_to_wide: false
+  }
+];
+
+export function getInstructionMatch(opcode) {
+  let match = _.find(INSTRUCTION_INDICES, (indices) => {
+    return (opcode >= indices.low_opcode_index &&
+            opcode <= indices.high_opcode_index);
+  });
+
+  if (match) {
+    return match;
+  }
+
+  return null;
+}
+
+export function createInstruction(idx, opcode, wide) {
+  let match = getInstructionMatch(opcode);
+  let instance = null;
+  if (match !== null) {
```

Comment:
Just to prevent nesting, what if you change the logic to have less nesting like the following?

```javascript
if (match === null) {
  return new AbstractInstruction(idx, opcode);
}

if (match.subject_to_wide) {
  instance = new match.type(idx, opcode, wide);
} else {
  instance = new match.type(idx, opcode);
}
if (match.wrap && match.wrap !== null) {
  instance = new match.wrap(instance);
}
```